src/http_client_async.zig

```diff
@@ -1464,6 +1499,16 @@ pub fn deinit(this: *HTTPClient) void {
         tunnel.deinit();
         this.proxy_tunnel = null;
     }
+
+    if (this.tls_props) |*tls_props| {
+        tls_props.deinit();
+        this.tls_props = null;
+    }
+
+    if (this.custom_context) |custom_context| {
+        bun.default_allocator.destroy(custom_context);
```

Comment:
where is the `us_socket_context_t` freed?

Reply:
@Jarred-Sumner https://github.com/oven-sh/bun/pull/6621/commits/b8d66e7edc45eee4bb606ed343dd93fadaae0828#diff-3563abb8107c1dcafabcb722b248f159cbefbaf4436cabcceb84e0e94f1d6041R362-R368